### DISCUSSION TITLE
Ability to catch asynchronous runtime errors

### DISCUSSION THREAD

**Discussion by AlexMath23** [saghul/txiki.js#663] [Ideas]
Hi, we are currently using [lvgljs](https://github.com/lvgl/lv_binding_js) with the txiki.js runtime. A nice certainty for us would to be to know that we can catch thrown runtime errors 100% of the time. But unfortunately in txiki.js we only have that certainty for the synchronous ones ( with [React Error Boundary).](https://www.npmjs.com/package/react-error-boundary) If 3rd party libraries throw async errors in promises without proper catch blocks those will slip by. In the browser / node you can handle these async, uncaught errors on a global level like this. 
![image](https://github.com/user-attachments/assets/67788d01-5472-45dd-8879-232b049012ca)
![image](https://github.com/user-attachments/assets/020cf6f4-24da-4874-b582-7d4ee6de6e51)
But this doesn't seem to be possible in txiki.js
To me it seems like a nice feature to have so you can have absolute protection from runtime errors / ways to recover in critical systems.

**Comment by saghul** [comment]
You can do this:

```js
window.addEventListener('unhandledrejection', event => {
    // TODO: handle an async rejection.
});
```

**Reply by AlexMath23** [reply]
hi @saghul , I already tried a few of these, none seem to work. 
![image](https://github.com/user-attachments/assets/e7dcd347-9ea5-488a-8f0c-81357a87e8bf)

**Reply by saghul** [reply]
Only the first one is implemented. I see that you really asked about exceptions, not rejections 😅 . I hope to implement the `error` event in a similar way.

**Reply by AlexMath23** [reply]
That's great to hear ! thanks for all the great work btw, I really like the runtime
